lib/createRef.lua

```diff
@@ -1,20 +1,38 @@
 --[[
-	Provides an API for acquiring a reference to a reified object. This
-	API is designed to mimic React 16.3's createRef API.
-
-	See:
-	* https://reactjs.org/docs/refs-and-the-dom.html
-	* https://reactjs.org/blog/2018/03/29/react-v-16-3.html#createref-api
+	A ref is nothing more than a binding with a special field 'current'
+	that maps to the getValue method of the binding
 ]]
+local Binding = require(script.Parent.Binding)
+
+local function createRef()
+	local binding = Binding.create(nil)
+
+	local ref = {}
+
+	--[[
+		A ref is just redirected to a binding via its metatable
+	]]
+	setmetatable(ref, {
+		__index = function(self, key)
+			if key == "current" then
+				return binding:getValue()
+			else
+				return binding[key]
+			end
+		end,
+		__newindex = function(self, key, value)
+			if key == "current" then
+				-- TODO: Confirm that this will not break any existing behavior
+				error("Assigning to ref.current is not allowed!", 2)
```

Comment:
This needs further consideration. There may be places elsewhere in our code where we assign to `current`.